gps/deduce.go

```diff
@@ -286,29 +283,11 @@ func (m gopkginDeducer) deduceSource(p string, u *url.URL) (maybeSource, error)
 		u.Path = path.Join(v[2], v[3])
 	}
 
-	unstable := false
-	majorStr := v[4]
-
-	if strings.HasSuffix(majorStr, gopkgUnstableSuffix) {
-		unstable = true
-		majorStr = strings.TrimSuffix(majorStr, gopkgUnstableSuffix)
-	}
-	major, err := strconv.ParseUint(majorStr[1:], 10, 64)
-	if err != nil {
-		// this should only be reachable if there's an error in the regex
-		return nil, fmt.Errorf("could not parse %q as a gopkg.in major version", majorStr[1:])
-	}
-
 	mb := make(maybeSources, len(gopkginSchemes))
```

Comment:
Why are we modifying a `github` url with `gopkginSchemes`?

Reply:
It's supposed to be a conservative approach, mirroring gopkg.in's current behavior most closely: the real, running service supports only `http` and `https`, not `ssh` or `git`.